homeassistant/components/screenlogic/__init__.py

```diff
@@ -1,143 +1,71 @@
 """The Screenlogic integration."""
 import asyncio
-from slugify import slugify
 from collections import defaultdict
 from datetime import timedelta
-import voluptuous as vol
 import logging
 
-from screenlogicpy import (
-    ScreenLogicGateway,
-    ScreenLogicError,
-)
+from screenlogicpy import ScreenLogicError, ScreenLogicGateway
 from screenlogicpy.const import (
     CONTROLLER_HARDWARE,
     SL_GATEWAY_IP,
-    SL_GATEWAY_PORT,
     SL_GATEWAY_NAME,
+    SL_GATEWAY_PORT,
 )
 
-from homeassistant import config_entries
 from homeassistant.config_entries import ConfigEntry
-import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PORT, CONF_SCAN_INTERVAL
 from homeassistant.core import HomeAssistant
-
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
 from homeassistant.helpers.update_coordinator import (
     CoordinatorEntity,
     DataUpdateCoordinator,
     UpdateFailed,
 )
 
-from homeassistant.const import (
-    CONF_IP_ADDRESS,
-    CONF_PORT,
-    CONF_SCAN_INTERVAL,
-    CONF_NAME,
-    CONF_HOST,
-)
-
-from .const import (
-    DOMAIN,
-    DEFAULT_SCAN_INTERVAL,
-    MIN_SCAN_INTERVAL,
-)
-
-from .config_flow import discover_gateways
+from .config_flow import async_discover_gateways_by_unique_id, name_for_mac
+from .const import DEFAULT_SCAN_INTERVAL, DISCOVERED_GATEWAYS, DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
-CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_IP_ADDRESS): cv.string,
-                vol.Optional(CONF_PORT, default=80): cv.positive_int,
-                vol.Optional(
-                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
-                ): vol.All(cv.positive_int, vol.Clamp(min=MIN_SCAN_INTERVAL)),
-            }
-        )
-    },
-    extra=vol.ALLOW_EXTRA,
-)
-
 PLATFORMS = ["switch", "sensor", "binary_sensor", "water_heater"]
 
 
 async def async_setup(hass: HomeAssistant, config: dict):
     """Set up the Screenlogic component."""
-    _LOGGER.info("Async Setup")
```

Comment:
I took out the YAML config since these aren't accepted anymore per https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md